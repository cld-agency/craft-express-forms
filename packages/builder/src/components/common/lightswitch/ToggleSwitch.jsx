import React from 'react';
import PropTypes from 'prop-types';
import './toggleSwitch.styl';

const ToggleSwitch = ({ value = false, toggleHandler }) => (
  <div
    className={`toggle-switch ${value ? 'on' : 'off'}`}
    onClick={toggleHandler}
    aria-checked={!!value}
    role="checkbox"
  >
    <div className="toggle-switch-handle" />
  </div>
);

ToggleSwitch.propTypes = {
  value: PropTypes.bool,
  toggleHandler: PropTypes.func,
};

export default ToggleSwitch;
