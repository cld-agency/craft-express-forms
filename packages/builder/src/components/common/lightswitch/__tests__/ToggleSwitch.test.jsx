import { shallow } from 'enzyme';
import React from 'react';
import ToggleSwitch from '../ToggleSwitch';

describe('<ToggleSwitch />', () => {
  it('renders & displays correctly', () => {
    const wrapper = shallow(<ToggleSwitch value={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Has the 'on' class when value is true", () => {
    const wrapper = shallow(<ToggleSwitch value={true} />);
    expect(wrapper.find('div.toggle-switch.on')).toHaveLength(1);
  });

  it("Does not have the 'on' class when value is false", () => {
    const wrapper = shallow(<ToggleSwitch value={false} />);
    expect(wrapper.find('div.toggle-switch.off')).toHaveLength(1);
  });

  it('changes value', () => {
    const onChange = jest.fn().mockReturnValue('clicked');
    const wrapper = shallow(<ToggleSwitch value={true} toggleHandler={onChange} />);

    const toggleSwitch = wrapper.find('div.toggle-switch.on');

    toggleSwitch.simulate('click');
    expect(onChange).toHaveBeenCalled();
  });
});
