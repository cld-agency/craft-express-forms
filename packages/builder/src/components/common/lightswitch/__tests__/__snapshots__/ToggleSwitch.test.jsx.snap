// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> renders & displays correctly 1`] = `
<div
  aria-checked={true}
  className="toggle-switch on"
  role="checkbox"
>
  <div
    className="toggle-switch-handle"
  />
</div>
`;
